### PR TITLE
update intl to latest version

### PR DESCRIPTION
**Pull request by JakeChampion**

### CODE REVIEW

**Comment by JakeChampion** [comment]
@caridy, would you be able to review this change?

**Comment by caridy** [comment]
Release notes: 

https://github.com/andyearnshaw/Intl.js/releases/tag/v1.2.1

File: polyfills/Intl/~locale/af-NA/config.json [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-	}
-}
+    "browsers": {
+        "android": "<=4.3",
+        "ie": "<11",
+        "ie_mob": "10",
+        "firefox": "<29",
+        "opera": "<15",
+        "chrome": "<24",
+        "safari": "*",
```
Comment (caridy) [inline polyfills/Intl/~locale/af-NA/config.json]: safari technical preview supports Intl, at some point we will have to add it here.

File: polyfills/Intl/~locale/af-NA/config.json [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+        "firefox_mob": "*"
+    },
+    "dependencies": [
+        "Intl"
+    ],
+    "build": {
+        "minify": false
+    },
+    "license": "MIT",
+    "spec": "http://www.ecma-international.org/ecma-402/1.0/",
```
Comment (caridy) [inline polyfills/Intl/~locale/af-NA/config.json]: https://tc39.github.io/ecma402/2016/ that's 3rd edition.

File: polyfills/HTMLPictureElement/polyfill.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
-/*! Picturefill - v3.0.1 - 2015-09-30
- * http://scottjehl.github.io/picturefill
- * Copyright (c) 2015 https://github.com/scottjehl/picturefill/blob/master/Authors.txt; Licensed MIT
+/*! picturefill - v3.0.2 - 2016-02-12
```
Comment (caridy) [inline polyfills/HTMLPictureElement/polyfill.js]: this is not related, restore this, and probably submit it in a different PR

File: polyfills/Intl/~locale/af-NA/config.json (hunk 1 above)
Comment (caridy) [inline polyfills/Intl/~locale/af-NA/config.json]: this probably need to be change in the build, since this is used by all files.

**Comment by caridy** [comment]
done with the first pass @JakeChampion

File: polyfills/Intl/~locale/af-NA/config.json (hunk 1 above)
Comment (JakeChampion) [inline polyfills/Intl/~locale/af-NA/config.json]: That's 9.1.1, correct?

Comment (caridy) [inline polyfills/Intl/~locale/af-NA/config.json]: @ericf can you help here? I think it is 9.1.1, but I'm not sure.

Comment (ericf) [inline polyfills/Intl/~locale/af-NA/config.json]: Right now the Safari TP is still using 9.x, but my guess is that the Safari version number will be 10 once this is released to the general public.

**Comment by caridy** [comment]
@JakeChampion we should hold this one for a bit since 1.2.2 will be released soon with some extra patches for some regressions.

**Comment by triblondon** [comment]
@JakeChampion I move to drop this from 3.6 so we can make a 3.6 release tomorrow without blockers.

**Comment by caridy** [comment]
@JakeChampion v1.2.2 is now in npm, we tested for few days, and seems to be ok.

**Comment by JakeChampion** [comment]
@caridy I can't install that version for some reason.

```
npm i intl
npm ERR! Darwin 15.4.0
npm ERR! argv "/Users/jake.champion/.nvm/versions/node/v6.1.0/bin/node" "/Users/jake.champion/.nvm/versions/node/v6.1.0/bin/npm" "i" "intl"
npm ERR! node v6.1.0
npm ERR! npm  v3.8.6

npm ERR! No compatible version found: intl@~1.2.2
npm ERR! Valid install targets:
npm ERR! 1.1.0, 1.0.1, 1.0.0, 1.0.0-rc-4, 1.0.0-rc-3, 1.0.0-rc-2, 1.0.0-rc-1, 0.1.4, 0.1.3, 0.1.2, 0.1.1, 0.1.0, 0.0.2, 0.0.1
npm ERR!
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/jake.champion/Code/work/origami/polyfill-service/npm-debug.log
```

**Comment by caridy** [comment]
https://www.npmjs.com/package/intl

> 1.2.2 is the latest of 17 releases
> published 3 days ago
> 6,289 downloads in the last day

Are you using a mirror or something?

**Comment by JakeChampion** [comment]
It turned out to be a caching issue. Resolved via `npm cache clean`.

File: polyfills/Intl/~locale/af-NA/config.json (hunk 1 above)
Comment (caridy) [inline polyfills/Intl/~locale/af-NA/config.json]: let's skip this for now, and wait until the official release to set this value.

**Comment by caridy** [comment]
the CI error doesn't seem to be related, but I might be wrong. In any case, this is looking good @JakeChampion.

**Comment by caridy** [comment]
Actually, we will need v1.2.3, what a mess! the issue with samsung browser was a real issue, and was patched here https://github.com/andyearnshaw/Intl.js/issues/170#issuecomment-220739146

**Comment by triblondon** [comment]
@caridy what should we do here?  Do we need another version bump?

We have a fair number of test failures currently:

```
 - ie 10 (Sauce results: https://saucelabs.com/tests/d5d23f1f751f4480a834a0568ed1fb68)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - ie 9 (Sauce results: https://saucelabs.com/tests/4370e2c0dfd14123a21076c211aff4cd)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - ie 8 (Sauce results: https://saucelabs.com/tests/00dd0c4925cb42079509c2dcad1ef6d3)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - ie 7 (Sauce results: https://saucelabs.com/tests/9792ab3740ac4b96b16ce8e3880d70d6)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - safari 9 (Sauce results: https://saucelabs.com/tests/0d1e35a8ed914a889ff31c201c36fdb5)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - safari 8 (Sauce results: https://saucelabs.com/tests/93c4c09755014529a8f08a2f8f2ba9c6)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
```

**Comment by caridy** [comment]
@triblondon let's wait until thru when I will release a new version that works on old IE and safari.

**Comment by triblondon** [comment]
@caridy can we progress this?

**Comment by JakeChampion** [comment]
@triblondon The latest version of Intl doesn't seem to pass our detect test on Safari 8-9 & IE 7-10.

**Comment by caridy** [comment]
@JakeChampion where can I see those failures? 

@triblondon we are still seeing issues with IE8, trying to find someone with IE8 and time to look at it. Here is the issue to track this https://github.com/andyearnshaw/Intl.js/issues/189

**Comment by JakeChampion** [comment]
This is what we get from CI:
<details>

```
Failed tests:
 - ie 10 (Sauce results: https://saucelabs.com/tests/adab4b0b7675494ea87b115718a13153)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - ie 9 (Sauce results: https://saucelabs.com/tests/f047a9adb85a457c9c06bbbb101ace2a)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - ie 8 (Sauce results: https://saucelabs.com/tests/4bd4de77741c4daab8fdba47e28e2738)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - ie 7 (Sauce results: https://saucelabs.com/tests/3ac14f4ed5e74cd890fa69592edd2fd3)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - safari 9 (Sauce results: https://saucelabs.com/tests/f7379671b2114fe0a5f54f2763d62e82)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
 - safari 8 (Sauce results: https://saucelabs.com/tests/c0b5ec4e9ad34785ac45dd5fd8c3d42b)
    -> Intl
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Intl
```

</details>

To see the results from a locally running version of the polyfill-service you can visit http://localhost/v2/test/tests?mode=all&feature=Intl and to view the current version in production you can visit https://cdn.polyfill.io/v2/test/tests?mode=all&feature=Intl

**Comment by JakeChampion** [comment]
This PR is now using the new third-party polyfill system which has resulted in a dramatic decrease in the amount of committed files :-)

**Comment by caridy** [comment]
I'm also planning to remove 52mb from Intl polyfill in the upcoming days.
